src/lib/mina_base/account.ml

```diff
@@ -590,6 +590,28 @@ let min_balance_at_slot ~global_slot ~cliff_time ~cliff_amount ~vesting_period
         | Some amt ->
             amt )
 
+let incremental_balance_between_slots ~start_slot ~end_slot ~cliff_time
+    ~cliff_amount ~vesting_period ~vesting_increment ~initial_minimum_balance :
+    Unsigned.UInt64.t =
+  let open Unsigned in
+  (* TODO/HACK: Archive table puts a bunch of 0s in timing_info table for accounts that don't vest. We should handle this more robustly. *)
+  if
+    Balance.(initial_minimum_balance = zero)
+    || Global_slot.(equal vesting_period zero)
```

Comment:
Is this special case required? `min_balance_at_slot` would return zero in these cases

Reply:
I think I was getting some division by zero errors. Threw this in as a hack and it helped but I'll look into it more carefully.